Sources/SKCore/FileBuildSettings.swift

```diff
@@ -29,3 +30,53 @@ public struct FileBuildSettings: Equatable {
     self.workingDirectory = workingDirectory
   }
 }
+
+fileprivate let headerExtensions = ["h", "hh", "hpp"]
+
+fileprivate let cExtensions = ["c"]
+fileprivate let cppExtensions = ["cpp", "cc"]
+fileprivate let objcExtensions = ["m"]
+fileprivate let objcppExtensions = ["mm"]
+
+private extension String {
+  var pathExtension: String {
+    return (self as NSString).pathExtension
+  }
+  var pathBasename: String {
+    return (self as NSString).lastPathComponent
+  }
+}
+
+public extension FileBuildSettings {
+  /// Return arguments suitable for use by `newFile`, which must be a header file.
+  ///
+  /// This patches the arguments by searching for the argument corresponding to `oldFile` and replacing it.
+  func patching(for newFile: String, from oldFile: String) -> FileBuildSettings {
+    guard headerExtensions.contains(newFile.pathExtension) else {
+      return self
+    }
+    var arguments = self.compilerArguments
+    let oldBasename = oldFile.pathBasename
+    let fileExtension = oldFile.pathExtension
+    if let index = arguments.lastIndex(where: {
+      // It's possible the arguments use relative paths while the `oldFile` given is
+      // an absolute/real path value. We guess based on suffixes instead of hitting
+      // the file system.
+      $0.hasSuffix(oldBasename) && oldFile.hasSuffix($0)
+    }) {
+      arguments[index] = newFile
+      // The `-x<lang>` flag needs to be before the possible `-c <header file>`
+      // argument in order for Clang to respect it.
```

Comment:
Probably worth mentioning in the comment that if there is an existing `-x` option, we will prefer that one since it will come later in the arguments (which seems like what you want anyway, if the user says `-x c++ foo.c` we would be better off with `-x c++` than with `-x c-header`

Reply:
Done